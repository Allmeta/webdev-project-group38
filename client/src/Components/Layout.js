import React, { Component } from 'react'
import styled from 'styled-components'
import Header from './Header'
import { Card } from 'semantic-ui-react'
import MovieCard from './Card.js'
import axios from 'axios'

const WrapperFlex = styled.div`
  display: flex;
  background-color: #212733;
  flex-direction: column;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
  color: white;
  height: 100vh;
  margin:0;
`
const StyledBody = styled.div`
  flex: 1;
  height: 100%;
  padding-top:20px;
  padding-left:10%;
  padding-right:10%;
`

class Layout extends Component {
  constructor (props) {
    super(props)
    this.state = {
      cards: []
    }
    this.image_base = 'https://image.tmdb.org/t/p/w500'
  }
  componentDidMount () {
    const instance = axios.create({
      baseURL: 'http://localhost:5025/api'
    })
    instance.get('/movies')
      .then((res) => {
        for (let i of res.data) {
          let str = '/genres/movie/id'.replace('id', i.movie_id)
          instance.get(str)
            .then((res2) => {
              i.genres = []
              for (let j of res2.data) {
                i.genres.push(j.name)
                // TODO set state
              }
            })
            .catch((err) => {
              console.log(err)
            })
        }
        this.setState({
          cards: res.data
        })
      })
      .catch((err) => {
        console.log(err)
      })
  }
  render () {
    return (
      <WrapperFlex>
        <Header />
        <StyledBody>
          <Card.Group>
            {this.state.cards.map((o) =>
              <MovieCard
                key={o.movie_id}
                img={this.image_base + o.poster_path}
                description={o.summary}
                title={o.title}
                popularity={Number(o.rating)}
                // genres={o.genres.join(' ')}
                genres='WIP'
              />
            )}
            {/* <MovieCard
              description="Lots of random latin words that have little meaning but to fill out descriptions like these."
              title="Lorem Ipsum"
              img={imeg}
              popularity={1234}
              genres="Action, Adventure, Thriller"/> */}

          </Card.Group>
        </StyledBody>
      </WrapperFlex >
    )
  }
}
export default Layout
