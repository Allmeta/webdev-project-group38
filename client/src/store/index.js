import { createStore, applyMiddleware, compose } from 'redux'
import { SearchFormReducer } from "../reducers/SearchFormReducer";
import thunk from 'redux-thunk'

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(SearchFormReducer, composeEnhancers(
  applyMiddleware(thunk)
));

export default store;