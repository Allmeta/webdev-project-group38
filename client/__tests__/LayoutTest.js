import React from 'react'
import Layout from '../src/Components/Layout.js'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { Provider } from 'react-redux'
import store from '../src/store/index.js'

it('renders correctly', () => {
  const tree = renderer.create(<Provider store={store}><Layout /></Provider>).toJSON()
  expect(tree).toMatchSnapshot()
})
