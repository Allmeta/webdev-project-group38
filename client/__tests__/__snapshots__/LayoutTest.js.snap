// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  height: 58px;
  left: 0;
  top: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  overflow-y: hidden;
  z-index: 999;
  background-color: #4553F4;
  box-shadow: 0 4px 3px rgba(49,54,68,.3);
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin-left: 10%;
  margin-right: 10%;
}

.c3 {
  width: 200px;
  border-right: 1px solid hsla(0,0%,93%,.16);
}

.c4 {
  width: 400px;
}

.c4 > * {
  margin-left: 50px;
}

.c6 {
  display: inline-grid;
  grid-template-columns: repeat(3,1fr);
  grid-column-gap: 30px;
  grid-row-gap: 30px;
  margin: 0 auto;
  max-width: 80vw;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-family: Roboto,sans-serif;
  color: white;
  min-height: 100vh;
  margin: 0;
  background: #edf1f5;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  padding: 108px 50px 50px 50px;
}

.c5 > * {
  margin-bottom: 50px;
}

@media (max-width:1330px) {
  .c6 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (max-width:860px) {
  .c6 {
    grid-template-columns: repeat(1,1fr);
    max-width: 100vw;
    grid-row-gap: 20px;
  }
}

@media (max-width:860px) {
  .c5 {
    padding: 78px 20px 20px 20px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <img
        className="c3"
        src="test-file-stub"
      />
      <div />
      <div
        className="c4"
      >
        <form
          className="ui form"
          onSubmit={[Function]}
        >
          <div
            className="field"
          >
            <div
              className="ui action input"
            >
              <input
                name="title"
                onChange={[Function]}
                placeholder="Search"
                type="text"
              />
              <button
                className="ui icon button"
                onClick={[Function]}
                role="button"
                type="submit"
              >
                <i
                  aria-hidden="true"
                  className="search icon"
                />
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    className="c5"
  >
    <div
      className="c6"
    />
    <i
      aria-hidden="true"
      className="grey arrow alternate circle down huge link icon"
      onClick={[Function]}
    />
  </div>
</div>
`;
